src/packages/galaxy/galaxy.py

```diff
@@ -271,6 +273,14 @@ def __init__(self, url: str = None):
                 [f"{key}={escape(str(value))}" for key, value in params.items()]
             )
         url = f"{base_url}{endpoint}?{param_string}"
-        async with aiohttp.ClientSession() as session:
-            async with session.get(url) as response:
-                return json.loads(await response.text())
+        retry_count = 0
+        while True:
+            try:
+                async with ClientSession(raise_for_status=True,
+                                         timeout=self.TIMEOUT) as session:
+                    async with session.get(url) as response:
+                        return json.loads(await response.text())
+            except ClientError:
+                retry_count += 1
+                if retry_count > self.MAX_RETRIES:
+                    raise
```

Comment:
[suggestion] following on above comment, this logic can probably be simplified, consider refactoring.

Reply:
Unfortunately I couldn't simplify this very well while also keeping the functionality I want. Which is to say, re-raising the exception from `aiohttp`.